### DISCUSSION TITLE
Format selection on YouTube

### DISCUSSION THREAD

**Discussion by thatsprettygood** [streamlink/streamlink#5787] [Q&A] [answered]
### Description

Is it possible to select a format? Choosing to use AV1, for example.

**Comment by WolfganP** [comment]
> Is it possible to select a format? Choosing to use AV1, for example.

When I need more format selection flexibility in youtube, I tend to use https://github.com/yt-dlp/yt-dlp/ and combine audio and video streams as wanted.

**Reply by bastimeyer** [reply]
Streamlink's YouTube plugin is limited in bandwidth for adaptive streams (basically all VOD content) anyway, as it doesn't implement the content signatures. So until someone implements it, it makes far more sense using different tools for downloading VOD content from YouTube, like yt-dlp.

**Comment by bastimeyer** [comment, marked as answer]
No, the plugin only yields a specific subset of stream codecs, defined by specific IDs:
https://github.com/streamlink/streamlink/blob/6.5.1/src/streamlink/plugins/youtube.py#L49

This isn't ideal and will be replaced in the future after Streamlink's new stream selection has landed:
#4902

However, as said, the YouTube plugin doesn't implement adaptive stream URL signatures, so the bandwidth is limited to ~50KiB/s, which pretty much makes watching YouTube VOD content using Streamlink impossible.

Whether YouTube offers AV1, H265 or VP9 live streams, I don't know, but those would be handled differently anyway.

**Reply by bastimeyer** [reply]
Here's some of the video data of this specific AV1 test video:
https://www.youtube.com/watch?v=m1jY2VLCRmY

```py
{133: ('240p', 'avc1.4d4015'),
 134: ('360p', 'avc1.4d401e'),
 135: ('480p', 'avc1.4d401e'),
 136: ('720p', 'avc1.4d401f'),
 137: ('1080p', 'avc1.640028'),
 160: ('144p', 'avc1.4d400c'),
 242: ('240p', 'vp9'),
 243: ('360p', 'vp9'),
 244: ('480p', 'vp9'),
 247: ('720p', 'vp9'),
 248: ('1080p', 'vp9'),
 271: ('1440p', 'vp9'),
 278: ('144p', 'vp9'),
 313: ('2160p', 'vp9'),
 394: ('144p', 'av01.0.00M.08'),
 395: ('240p', 'av01.0.00M.08'),
 396: ('360p', 'av01.0.01M.08'),
 397: ('480p', 'av01.0.04M.08'),
 398: ('720p', 'av01.0.05M.08'),
 399: ('1080p', 'av01.0.08M.08'),
 400: ('1440p', 'av01.0.12M.08'),
 401: ('2160p', 'av01.0.12M.08'),
 571: ('4320p', 'av01.0.16M.08')}
```
